lib/repository/cat/cat_repository_impl.dart

```diff
@@ -1,22 +1,20 @@
+import 'package:flutter_template/repository/cat/cat_repository.dart';
 import 'package:flutter_template/repository/cat/domain_cat_mapper.dart';
-import 'package:flutter_template/services/cat/remote/cat_remote_service.dart';
 import 'package:flutter_template/services/entity/cat/remote/remote_cat.dart';
 
-class CatRepositoryImpl implements CatRemoteService {
-  late final CatRemoteService catRemoteService;
-
-  late final DomainCatMapper domainCatMapper;
+class CatRepositoryImpl implements CatRepository {
+  final CatRepository catRepository;
+  final DomainCatMapper domainCatMapper;
 
   CatRepositoryImpl({
-    required this.catRemoteService,
+    required this.catRepository,
     required this.domainCatMapper,
   });
 
   @override
-  Future<List<RemoteCat>> searchCat({required String searchTerm}) async {
-    final results = await catRemoteService.searchCat(
-      searchTerm: searchTerm,
-    );
-    return domainCatMapper.map(results);
+  Future<List<RemoteCat>> searchCat(String searchTerm) async {
+    final result = await catRepository.searchCat(searchTerm);
+
+    return result;
   }
```

Comment:
Same here. This is basically a recursion with no exit.

Reply:
There is an error that is coming up with this. I've kept this for later fix.